diff --git a/node_modules/easymde/src/js/easymde.js b/node_modules/easymde/src/js/easymde.js
index cece3b0..48968dd 100644
--- a/node_modules/easymde/src/js/easymde.js
+++ b/node_modules/easymde/src/js/easymde.js
@@ -855,18 +855,22 @@ function drawLink(editor) {
 /**
  * Action for drawing an img.
  */
-function drawImage(editor) {
+function drawImage(editor,uuu,capt) {
     var cm = editor.codemirror;
     var stat = getState(cm);
     var options = editor.options;
-    var url = 'https://';
-    if (options.promptURLs) {
+    var url = (uuu||'https://');
+    if (options.promptURLs && typeof uuu === 'undefined') {
         url = prompt(options.promptTexts.image, 'https://');
         if (!url) {
             return false;
         }
     }
-    _replaceSelection(cm, stat.image, options.insertTexts.image, url);
+    let insertTexts = options.insertTexts.image;
+    if(typeof capt === 'string') {
+        insertTexts = ['<figure class="image image_resized"><img src="','#url#" /><figcaption>'+ capt +'</figcaption></figure>',]
+    }
+    _replaceSelection(cm, stat.image, insertTexts, url);
 }
 
 /**
@@ -2863,8 +2867,8 @@ EasyMDE.prototype.cleanBlock = function () {
 EasyMDE.prototype.drawLink = function () {
     drawLink(this);
 };
-EasyMDE.prototype.drawImage = function () {
-    drawImage(this);
+EasyMDE.prototype.drawImages = function (url,capt) {
+    drawImage(this,url,capt);
 };
 EasyMDE.prototype.drawUploadedImage = function () {
     drawUploadedImage(this);
diff --git a/node_modules/easymde/types/easymde.d.ts b/node_modules/easymde/types/easymde.d.ts
index 2aa161b..8a495fe 100644
--- a/node_modules/easymde/types/easymde.d.ts
+++ b/node_modules/easymde/types/easymde.d.ts
@@ -254,6 +254,7 @@ declare class EasyMDE {
 
     clearAutosavedValue(): void;
 
+    drawImages: (url: string) => void;
     static toggleBold: (editor: EasyMDE) => void;
     static toggleItalic: (editor: EasyMDE) => void;
     static toggleStrikethrough: (editor: EasyMDE) => void;
