diff --git a/node_modules/@strapi/strapi/lib/commands/develop.js b/node_modules/@strapi/strapi/lib/commands/develop.js
index 23f25b7..1bd40a8 100644
--- a/node_modules/@strapi/strapi/lib/commands/develop.js
+++ b/node_modules/@strapi/strapi/lib/commands/develop.js
@@ -66,6 +66,7 @@ const primaryProcess = async ({ distDir, appDir, build, isTSProject, watchAdmin,
         srcDir: appDir,
       });
     } catch (err) {
+      console.error(err)
       process.exit(1);
     }
   }
@@ -76,6 +77,7 @@ const primaryProcess = async ({ distDir, appDir, build, isTSProject, watchAdmin,
         stdio: 'inherit',
       });
     } catch (err) {
+      console.error(err)
       process.exit(1);
     }
   }
diff --git a/node_modules/@strapi/strapi/lib/core-api/controller/transform.js b/node_modules/@strapi/strapi/lib/core-api/controller/transform.js
index af078c3..f0f69c9 100644
--- a/node_modules/@strapi/strapi/lib/core-api/controller/transform.js
+++ b/node_modules/@strapi/strapi/lib/core-api/controller/transform.js
@@ -35,8 +35,7 @@ const transformComponent = (data, component) => {
     return res;
   }
 
-  const { id, attributes } = res;
-  return { id, ...attributes };
+  return res
 };
 
 const transformEntry = (entry, type) => {
@@ -63,9 +62,12 @@ const transformEntry = (entry, type) => {
     if (attribute && attribute.type === 'relation') {
       const data = transformEntry(property, strapi.contentType(attribute.target));
 
-      attributeValues[key] = { data };
+      if(Array.isArray(data)) attributeValues[key] = data;
+      else attributeValues[key] = { ...data };
     } else if (attribute && attribute.type === 'component') {
-      attributeValues[key] = transformComponent(property, strapi.components[attribute.component]);
+      const data = transformComponent(property, strapi.components[attribute.component]);
+      if(Array.isArray(data)) attributeValues[key] = data;
+      else attributeValues[key] = data
     } else if (attribute && attribute.type === 'dynamiczone') {
       if (isNil(property)) {
         attributeValues[key] = property;
@@ -77,7 +79,7 @@ const transformEntry = (entry, type) => {
     } else if (attribute && attribute.type === 'media') {
       const data = transformEntry(property, strapi.contentType('plugin::upload.file'));
 
-      attributeValues[key] = { data };
+      attributeValues[key] = { ...data };
     } else {
       attributeValues[key] = property;
     }
@@ -85,7 +87,7 @@ const transformEntry = (entry, type) => {
 
   return {
     id,
-    attributes: attributeValues,
+    ...attributeValues,
     // NOTE: not necessary for now
     // meta: {},
   };
diff --git a/node_modules/@strapi/strapi/lib/types/core/strapi/index.d.ts b/node_modules/@strapi/strapi/lib/types/core/strapi/index.d.ts
index 5e07ae1..8b98c6d 100644
--- a/node_modules/@strapi/strapi/lib/types/core/strapi/index.d.ts
+++ b/node_modules/@strapi/strapi/lib/types/core/strapi/index.d.ts
@@ -351,11 +351,6 @@ export interface Strapi {
    * Entity Validator instance
    */
   entityValidator: any;
-
-  /**
-   * Entity Service instance
-   */
-  entityService: any;  
 }
 
 export interface Lifecycles {
